public_html/wp-content/plugins/wc-post-types/inc/rest-api.php

```diff
@@ -147,8 +155,60 @@ function register_additional_rest_fields() {
 		]
 	);
 }
+
 add_action( 'rest_api_init', __NAMESPACE__ . '\register_additional_rest_fields' );
 
+/**
+ * Validate simple meta query parameters in an API request and add them to the args passed to WP_Query.
+ *
+ * @param array $args    The prepared args for the WP_Query object.
+ * @param array $request The args from the REST API request.
+ *
+ * @return array
+ */
+function prepare_meta_query_args( $args, $request ) {
+	if ( isset( $request['meta_key'], $request['meta_value'] ) ) {
+		$args['meta_key']   = $request['meta_key'];
+		$args['meta_value'] = $request['meta_value'];
+	}
+
+	return $args;
+}
+
+add_filter( 'rest_wcb_session_query', __NAMESPACE__ . '\prepare_meta_query_args', 10, 2 );
```

Comment:
> These request parameters are not available for any post type endpoints out of the box, presumably for security reasons.

🤔 

The *fields* aren't included in the response by default, [because of security concerns](https://github.com/WP-API/WP-API/pull/135), but it does seem a bit odd that the `meta_key`/`meta_value` *parameters* aren't enabled by default. I would have thought they'd been enabled, but would ignore any data that wasn't registered with `show_in_rest => true`.

I wonder if there's some reason we shouldn't enable them, or need to be more cautious when using them? 

It feels weird that we need to enable those params, rather than just registering the data, since exposing that data is a common use case, and it'd be really weird to require all devs to enable those fields any time they want to do something simple like this. 

- [ ] `SHOULD` - This feels like it might need a bit more research, unless you've already done that and found something to indicate this is the recommended way. If it is necessary, I imagine there's an existing ticket complaining about it :)

- [ ] `SHOULD` - At the very least, we should include a link to that ticket as a `@todo` note so we can remove this extra code when Core provides a better way to do it.

- [ ] `SHOULD` - If that ticket doesn't exist, it seems like it'd be good to create it :)

Reply:
Before the API was merged into Core, meta queries were possible via a `filter` parameter:

```
?filter[meta_key]=foo&filter[meta_value]=bar
```

This was [removed](https://core.trac.wordpress.org/ticket/38378) when the API was merged. There ~is~ was [a plugin](https://github.com/WP-API/rest-filter) that restored this functionality, but that doesn't seem like a good long term solution. All discussions I have found so far, though, just recommend installing that plugin. My Trac-fu might be failing me, but I can't find any Core tickets about adding the ability to do meta queries.

I'm waiting to hear back in the `core-restapi` channel to see if anyone there has insight.

Reply:
Here's a relevant discussion:
https://wordpress.slack.com/archives/C02RQC26G/p1532356391000058

Reply:
Here's a conversation thread about this with Kadam:
https://wordpress.slack.com/archives/C02RQC26G/p1557261445107000

The gist:

> It's possible to use meta queries to construct a very expensive DB query, even with the hard limit of 100 posts_per_page, so in an abundance of caution we restricted passing meta queries via the REST API query for the API beta and subsequent initial core release of the plugin
> It has since become appealing to add it back in, with the improvements to meta registration you mention and the introduction of register_post_meta and friends, but it hasn't been a widespread enough request yet to prioritize

Reply:
👍 

Sounds like we can move forward with the current approach then, but it'd be nice to add a `todo` with a link to the ticket, once you create it.

Reply:
Ok, here's a Trac ticket: https://core.trac.wordpress.org/ticket/47194

I'm thinking we should namespace our custom collection params since we don't know what, if anything, a Core implementation would look like.

Reply:
c024005856199dbdbb19ba5ab850d1a35eb5eb5d

Reply:
👍